vm/interface.go

```diff
@@ -0,0 +1,66 @@
+package vm
+
+import (
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"math/big"
+)
+
+type ExecuteArguments struct {
+	Sender   []byte
+	Value    *big.Int
+	Function string
+	Args     []*big.Int
+}
+
+// SystemSmartContract interface defines the function a system smart contract should have
+type SystemSmartContract interface {
+	Execute(args *ExecuteArguments) vmcommon.ReturnCode
+	ValueOf(key interface{}) interface{}
+	IsInterfaceNil() bool
+}
+
+// SystemSCContainerFactory defines the functionality to create a system smart contract container
+type SystemSCContainerFactory interface {
+	Create() (SystemSCContainer, error)
+	IsInterfaceNil() bool
+}
+
+// SystemSCContainer defines a system smart contract holder data type with basic functionality
+type SystemSCContainer interface {
+	Get(key []byte) (SystemSmartContract, error)
+	Add(key []byte, val SystemSmartContract) error
+	AddMultiple(keys [][]byte, scs []SystemSmartContract) error
+	Replace(key []byte, val SystemSmartContract) error
+	Remove(key []byte)
+	Len() int
+	Keys() [][]byte
+	IsInterfaceNil() bool
+}
+
+// SystemEI defines the environment interface system smart contract can use
+type SystemEI interface {
+	Transfer(destination []byte, sender []byte, value *big.Int, input []byte) error
+	BlockHash(number int64) []byte
+	GetVMInput() vmcommon.VMInput
```

Comment:
This is a minimal number of functions the SC need to interact with. In case of ethereum is even bigger, for Arwen is big as well. Should the SC know about a number of interfaces ? It is easier for it to interact with only one. This is the basic library, for specific functionalities we will have a new EI.